synthesizer/src/vm/finalize.rs

```diff
@@ -1875,115 +1939,1009 @@ finalize compute:
+
+        // Construct the bonded balances.
+        let bonded_balances = validators
+            .iter()
+            .map(|(private_key, (amount, _))| {
+                let address = Address::try_from(private_key).unwrap();
+                (address, (address, *amount))
+            })
+            .chain(delegators.iter().map(|(private_key, (validator, amount))| {
+                let address = Address::try_from(private_key).unwrap();
+                (address, (*validator, *amount))
+            }))
+            .collect::<IndexMap<_, _>>();
+
+        println!("Generating the genesis block.");
+
+        let genesis = vm
+            .genesis_quorum(
+                validators.keys().next().unwrap(),
+                committee.clone(),
+                public_balances.clone(),
+                bonded_balances.clone(),
+                rng,
+            )
+            .unwrap();
+
+        println!("Adding the genesis block to the VM.");
+
+        // Add the genesis block to the VM.
+        vm.add_next_block(&genesis).unwrap();
+
+        // Check that the state of the `credits.aleo` program is correct.
+        let program_id = ProgramID::from_str("credits.aleo").unwrap();
+        let committee_mapping_name = Identifier::from_str("committee").unwrap();
+        let account_mapping_name = Identifier::from_str("account").unwrap();
+        let bonded_mapping_name = Identifier::from_str("bonded").unwrap();
+        let metadata_mapping_name = Identifier::from_str("metadata").unwrap();
+        let unbonding_mapping_name = Identifier::from_str("unbonding").unwrap();
+
+        // Get the committee mapping.
+        let actual_committee = vm.finalize_store().get_mapping_confirmed(program_id, committee_mapping_name).unwrap();
+        let expected_committee = committee
+            .members()
+            .iter()
+            .map(|(address, (amount, is_open))| {
+                (
+                    Plaintext::from_str(&address.to_string()).unwrap(),
+                    Value::from_str(&format!("{{ microcredits: {amount}u64, is_open: {is_open} }}")).unwrap(),
+                )
+            })
+            .collect_vec();
+        assert_eq!(actual_committee.len(), expected_committee.len());
+        for entry in actual_committee.iter() {
+            assert!(expected_committee.contains(entry));
+        }
+
+        // Get the account mapping.
+        let actual_account = vm.finalize_store().get_mapping_confirmed(program_id, account_mapping_name).unwrap();
+        let expected_account = public_balances
+            .iter()
+            .map(|(address, amount)| {
+                (Plaintext::from_str(&address.to_string()).unwrap(), Value::from_str(&format!("{amount}u64")).unwrap())
+            })
+            .collect_vec();
+        assert_eq!(actual_account.len(), expected_account.len());
+        // Check that all entries except for the one for the first validator are the same.
```

Comment:
Fix comment. Also why is the first validator entry different? 

```suggestion
        // Check that all entries except for the first validator are the same.
```

Reply:
This is because the first validator also executes a number of additional transactions in the `genesis_quorum` method. 
Consequently, its balance after adding the genesis block is not the same as the public balance provided in `Genesis::Ratify`.

Reply:
Could you maybe clarify that in the comment? Otherwise, we won't know why down the line.